java-checks/src/test/files/checks/AssertOnBooleanVariableCheck.java

```diff
@@ -1,22 +1,45 @@
+import java.util.Collections;
+import java.util.List;
 
 public class A {
-  boolean removed;
-  void foo() {
-    List myList;
-    assert myList.remove(myList.get(0)); // Noncompliant [[sc=12;ec=40]] {{Move this "assert" side effect to another statement.}}
-    assert(myList.remove(myList.get(0))); // Noncompliant
-    this.removed = myList.remove(myList.get(0));
-    assert this.removed;
+  void foo(A a) {
+    List<Object> myList;
+    assert myList.remove(myList.get(0)); // Noncompliant [[sc=19;ec=25]] {{Move this "assert" side effect to another statement.}}
+    assert myList.remove(myList.remove(1)); // Noncompliant [[sc=19;ec=25]] {{Move this "assert" side effect to another statement.}}
+
+    assert myList.add(new Object()); // Noncompliant
+    assert myList.retainAll(Collections.singleton(new Object())); // Noncompliant
+    assert deleteStuff(); // Noncompliant
+    assert stuffToRemove(); // Compliant - does not start with 'remove'
+    assert updateIfValid(false); // Noncompliant
+    assert setValue(); // Noncompliant
+
+    assert bar() > 1 ? doNothing() : deleteStuff(); // Noncompliant [[sc=38;ec=49]]
+
+    assert bar() == 0; // Compliant
+    assert new A(){ // Compliant
+      @Override
+      boolean deleteStuff() {
+        // do nothing
+      }
+    }.bar() != null;
+
     boolean removed = myList.remove(myList.get(0));
     assert removed;
-    assert(removed);
+
     int i = 0;
-    assert (removed = i); // Noncompliant
-    assert bar() == 1; // Noncompliant
-    assert i++; // Noncompliant
+    assert (removed = i); // Compliant - no call to side-effect method
+    assert i++; // Compliant
```

Comment:
This doesn't compile, `i++` is not boolean expr, also L31 doesn't compile for same reason